src/test/java/com/wikia/webdriver/testcases/discussions/PollsTests.java

```diff
@@ -0,0 +1,158 @@
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.BasePostsCreator;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.Poll;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.SignInToFollowModalDialog;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.TextGenerator;
+import com.wikia.webdriver.elements.mercury.pages.discussions.PostsListPage;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+@Test(groups = "discussions-polls")
+public class PollsTests extends NewTestTemplate {
+
+    public static int DEFAULT_ANSWERS_NUMBER = 2;
+
+    @Test
+    @InBrowser(emulator = Emulator.DESKTOP_BREAKPOINT_BIG)
+    @Execute(asUser = User.USER_6, onWikia = MercuryWikis.DISCUSSIONS_2)
+    public void userCanCreatePostWithSimplePollOnDesktop() {
+        PostsListPage page = new PostsListPage().open();
+        BasePostsCreator postsCreator = page.getPostsCreatorDesktop();
+        postsCreator.click().closeGuidelinesMessage().clickAddCategoryButton().selectFirstCategory();
+        postsCreator.addTitleWith(TextGenerator.createUniqueText());
+        Poll poll = postsCreator.addPoll();
+        Assert.assertFalse(postsCreator.isPostButtonActive());
+
+        poll.addTitle(TextGenerator.createUniqueText());
+        Assert.assertFalse(postsCreator.isPostButtonActive());
+        poll.addNthAnswer(TextGenerator.createUniqueText(), 0);
+        Assert.assertFalse(postsCreator.isPostButtonActive()); //poll needs to have at least 2 answers options
+        poll.addNthAnswer(TextGenerator.createUniqueText(), 1);
+        postsCreator.clickSubmitButton();
+        page.waitForLoadingSpinner();
+
+        Assert.assertTrue(new PostsListPage().getPost().firstPostHasPoll());
+    }
+
+    @Test(dependsOnMethods = {"userCanCreatePostWithSimplePollOnDesktop"})
+    @InBrowser(emulator = Emulator.DESKTOP_BREAKPOINT_BIG)
+    @Execute(asUser = User.ANONYMOUS, onWikia = MercuryWikis.DISCUSSIONS_2)
+    public void anonUserCanNotVoteInPollOnDesktop() {
+        Poll poll = new PostsListPage().open().getPost().clickNthPostWithPoll(0).getPoll();
+        poll.clickPollTitle();
+        SignInToFollowModalDialog signInModal = new SignInToFollowModalDialog();
+        Assert.assertTrue(signInModal.isVisible());
+
+        signInModal.clickOkButton();
+        poll.clickNthAnswer(0);
+        Assert.assertTrue(signInModal.isVisible());
+    }
+
+    @Test(dependsOnMethods = {"userCanCreatePostWithSimplePollOnDesktop"})
+    @InBrowser(emulator = Emulator.DESKTOP_BREAKPOINT_BIG)
+    @Execute(asUser = User.USER_3, onWikia = MercuryWikis.DISCUSSIONS_2)
+    public void loggedInUserCanVoteOnceInPollOnDesktop() {
+        Poll poll = new PostsListPage().open().getPost().clickNthPostWithPoll(0).getPoll();
+        Assert.assertTrue(poll.getAnswersRadioButtonsList().size() > 0);
```

Comment:
No need to check for poll validity in this test - this one should only verify voting part